docs/design/third-party-libraries.md

```diff
@@ -0,0 +1,175 @@
+# Installing third party libraries
+
+# Abstract
+
+The current `ng install` process is faulty; third parties have to add a `bundles/` directory
+with metadata that we expect. This document explores ways to improve on the process.
+
+# Requirements
+
+The following use cases need to be supported by `ng install`:
+
+1. Support for _any_ thirdparties, ultimately falling back to running `npm install` only and
+doing nothing else, if necessary.
+1. Not a new package manager.
+1. Metadata storage by thirdparties in `package.json`. This must be accessible by plugins for
+the build process or even when scaffolding.
+1. Proper configuration of SystemJs in the browser.
+1. SystemJs configuration needs to be kept separated from the autogenerated part.
+1. The build process right now is faulty because the `tmp/` directory used by Broccoli is
+inside the project, which means that `node_modules/` is understood by Typescript. We need to
+proper map the imports and move them to `dist/`.
+1. Potentially hook into scaffolding.
+1. Potentially hook into the build.
+1. Safe uninstall path.
+
+# Usages
+
+Here's a few stories that should work right off:
+
+```sh
+$ ng new my-app && cd my-app/
+$ ng install jquery
+```
+^(this makes jQuery available in the index)
+
+```sh
+$ ng install angularfire2
+> Please specify your Firebase database URL [my-app.firebaseio.com]: _
+```
+^(makes firebase available and provided to the App)
+
+```sh
+$ ng install angularfire2 --dbUrl=my-firebase-db.example.com
+```
+^(skip prompts for values passed on the command line)
+
+```sh
+$ ng install angularfire2 --quiet
+```
+^(using `--quiet` to skip all prompts and use default values)
+
+```sh
+$ ng install less
```

Comment:
With `ng install` we can install `less/sass/stylus` and enable the correlated CLI addon. I don't any see any repeating pattern on this three preprocessors so I think this will have to be hardcoded.